bluetooth_mesh/application.py

```diff
@@ -719,13 +724,10 @@ def _convert_config(self, configuration):
         iv_index: int = 0,
         addr: int = None,
         flags: Optional[Mapping[str, Any]] = None,
-    ) -> int:
+    ) -> asyncio.Future:
```

Comment:
I'd rather leave the public API as-is. Changes in `interfaces` are fine, but here I'd `await` for the join_complete future and return the received token.

Same goes for `join` method.